ocs_ci/ocs/hsbench.py

```diff
@@ -134,35 +152,202 @@ def run_benchmark(
-        num_objects = self.toolbox.exec_sh_cmd_on_pod(
-            f"radosgw-admin bucket stats --bucket={self.bucket_name} | grep num_objects"
+        cmd = f"cp {self.pod_obj.name}:/go/{result} " f"{self.warp_dir}/{result}"
+        self.ocp_obj.exec_oc_cmd(
+            command=cmd,
+            out_yaml_format=False,
+            timeout=180,
         )
-        if num_objects is not self.num_obj:
-            log.info(
-                f"Number object is created by hsbench in {self.bucket_name} "
-                f"is {num_objects}"
+        if os.path.getsize(f"{self.warp_dir}/{result}") != 0:
+            log.info("Workload was running...")
+        else:
+            raise UnexpectedBehaviour(
+                f"Output file {result} is empty, "
+                "Warp workload doesn't run as expected..."
+            )
+
+    def validate_s3_objects(self, upgrade=None):
+        """
+        Validate S3 objects using 'radosgw-admin' on single bucket
+        Validate objects in buckets after completed upgrade
+
+        Agr:
+            upgrade (str): Upgrade status
+        Raise:
+            UnexpectedBehaviour: If objects pre-upgrade and post-upgrade are not identical.
+
+        """
+        # self.upgrade_stat = upgrade_stat if upgrade_stat else self.upgrade_stat
+        for i in range(self.num_bucket):
+            bucket_name = self.bucket_prefix + "00000000000" + str(i)
+            num_objects = self.toolbox.exec_sh_cmd_on_pod(
+                f"radosgw-admin bucket stats --bucket={bucket_name} | grep num_objects"
+            )
+            if num_objects is not self.num_obj:
+                log.info(
+                    f"Number object is created by hsbench in {bucket_name} "
+                    f"is {num_objects}"
+                )
+            assert num_objects, (
+                "Number objects in bucket don't match."
+                f"Expecting {self.num_obj} but getting {num_objects}"
             )
-        assert num_objects, (
-            "Number objects in bucket don't match."
-            f"Expecting {self.num_obj} but getting {num_objects}"
+            # Save objects to a file for validation
+            file_path = f"{self.warp_dir}/obj_{upgrade}_{bucket_name}"
+            object_list = self.toolbox.exec_sh_cmd_on_pod(
+                f"radosgw-admin bi list --bucket={bucket_name}"
+            )
+            f = open(file_path, "w")
+            f.write(object_list)
+
+        # Validate objects in buckets for post upgrade
+        if upgrade == "post_upgrade" and bucket_name != "new000000000000":
+            for i in range(self.num_bucket):
+                if os.path.exists(f"{self.warp_dir}/obj_{upgrade}_{bucket_name}"):
+                    log.info(
+                        f"Verifying objects in bucket {bucket_name} for post-upgrade..."
+                    )
+                    if filecmp.cmp(
+                        f"{self.warp_dir}/obj_pre_upgrade_{bucket_name}",
```

Comment:
when was this directory got populated?

Reply:
Object contents are saved to a file here: https://github.com/red-hat-storage/ocs-ci/blob/1332ebb63dd97a2a51c7556fb8221f959740e2ee/ocs_ci/ocs/hsbench.py#L231